cEP0002.md

```diff
@@ -0,0 +1,161 @@
+#Next Gen Bear Design
+
+| Metadata     |                                         |
+| ------------ |-----------------------------------------|
+| cEP          | 2                                       |
+| Version      | 0.1                                     |  
+| Title        | Next gen bear design                    |
+| Authors      | Udayan Tandon <udayan12167@iiitd.ac.in> |
+| Status       | Proposed                                |
+| Type         | Process                                 |
+
+#Abstract
+
+This document describes how the next generation bears will be designed and 
+implemented.
+
+This cEP describes a new implementation of bears in coala which should make
+ them simpler and make their design more intuitive and better.
+
+#Introduction
+
+Currently the bear design is clunky and bad. The current design makes global 
+bears dependant on local bears through it's queuing mechanism. It uses 
+communication signals over a queue to indicate that global bears are done with 
+their execution and local bears can begin. The queueing mechanism doesn't work 
+very well as it sometimes has synchronization problems. Also if local bears 
+fail to run it stalls the global bears. Also we have a permanent logger thread 
+that we can do away with. An extra logger thread consumes time and resources 
```

Comment:
"do away"? That sounds wrong to me, you use it in multiple places. Maybe we should activate LanguageToolBear or something like it :P